classes/query/QueryDAO.php

```diff
@@ -271,36 +273,35 @@ public function deleteObject($query)
     /**
      * Delete a submission query by ID.
      *
+     * Deletes any associated notes and notifications. The
+     * participants will be deleted automatically through
+     * the onDelete CASCADE foreign key relationship in
+     * the database table.
+     *
      * @param int $queryId Query ID
      * @param int $assocType Optional ASSOC_TYPE_...
      * @param int $assocId Optional assoc ID per assocType
      */
     public function deleteById($queryId, $assocType = null, $assocId = null)
     {
-        $queryQuery = DB::table('queries')
-            ->where('query_id', '=', $queryId);
+        $noteDao = DAORegistry::getDAO('NoteDAO'); /** @var NoteDAO $noteDao */
+        $noteDao->deleteByAssoc(Application::ASSOC_TYPE_QUERY, $queryId);
```

Comment:
If the `assocType` and `assocId` are being specified as a safety mechanism (e.g. to ensure that a user-specified `queryId` belongs to an expected association) then deleting the note up front without checking the association info may permit malicious users to cause damage. I'd suggest deleting the note content *after* the query, and using the number of deleted rows in the query deletion to determine whether or not to execute this.

Reply:
Ok, I'm implemented it that way. It will break if/when we add foreign keys to the notes or notifications table, but a problem for later. :)